server/contexts/ctxerr/ctxerr.go

```diff
@@ -25,6 +26,31 @@ type key int
 
 const errHandlerKey key = 0
 
+// Cause returns the root error in err's chain.
+func Cause(err error) error {
```

Comment:
Looks great, we should add a small unit test for `ctxerr.Cause`, to be extra sure.

Reply:
:+1: added.